### PR TITLE
Fix bug with self closing tags.

### PR DESCRIPTION
**Pull request by miladkdz**

### CODE REVIEW

File: src/incubator-tooltip.html [hunk 1]
```diff
@@ -106,6 +106,11 @@
 
             _targetElement: {
               type: Object
+            },
+
+            _moved: {
```
Comment (peancored) [inline src/incubator-tooltip.html]: the variable name is not that clear

File: src/incubator-tooltip.html [hunk 2]
```diff
@@ -129,7 +134,10 @@
           }
           if (this._targetElement && !this.isAttached) {
             this.isAttached = true;
-            Polymer.dom(this._targetElement).appendChild(this);
+            if (['input', 'img'].indexOf(this._targetElement.tagName.toLowerCase()) === -1) {
```
Comment (peancored) [inline src/incubator-tooltip.html]: please extract the condition to a boolean variable to improve readability. something like `targetIsEmptyTag`, according to this 
https://www.w3schools.com/html/html_elements.asp

File: src/incubator-tooltip.html [hunk 3]
```diff
@@ -163,17 +176,30 @@
...
+            this.removeEventListener('mouseleave', this._boundHide);
+          }
         }
 
         _setPosition() {
           if (!this.isAttached) {
             return;
           }
-          const parentRect = this.offsetParent.getBoundingClientRect();
+          let parentRectTop = 0,
```
Comment (peancored) [inline src/incubator-tooltip.html]: What is the point of making the logic more complicated?

Comment (peancored) [reply src/incubator-tooltip.html]: if it is absolutely necessary pls comment that.

File: src/incubator-tooltip.html [hunk 4]
```diff
@@ -163,17 +176,30 @@
...
+            parentRectLeft = 0,
+            parentRectHeight = 0;
+          if (this.offsetParent) {
+            parentRectTop = this.offsetParent.getBoundingClientRect().top;
```
Comment (peancored) [inline src/incubator-tooltip.html]: you can request for bounding client rect once and store it in a variable

File: demo/incubator-tooltip-basic-demos.html [hunk 5]
```diff
@@ -21,9 +21,10 @@ <h1 id="to-be-tooltiped">
     <h3>Rich</h3>
     <vaadin-demo-snippet id="incubator-tooltip-demo-sample-2">
       <template preserve-content>
-        <h1 id="to-be-tooltiped">
+        <!-- <h1 id="to-be-tooltiped">
```
Comment (peancored) [inline demo/incubator-tooltip-basic-demos.html]: should it be removed?

File: src/incubator-tooltip.html (hunk 3 above)
Comment (miladkdz) [reply src/incubator-tooltip.html]: offsetParent is null for empty tags.

**Pull request merged**